oauthenticator/auth0.py

```diff
@@ -55,14 +58,21 @@ def _auth0_subdomain_default(self):
         return subdomain
 
     username_key = Unicode(
-        os.environ.get("OAUTH2_USERNAME_KEY", "email"),
         config=True,
-        help="Userdata username key from returned json with user data login information",
+        help="Deprecated, use `Auth0OAuthenticator.username_claim`",
     )
```

Comment:
## Important: breaking change

There is a breaking change related to the `OAUTH2_USERNAME_KEY` environment variable. Adding this line back, setting the default value of this deprecated trait, is not sufficient though.

Our logic associated with the configuration of `_deprecated_oauth_aliases` is working if we via config directly set `username_key`, but not if we set it indirectly via the default value via the `OAUTH2_USERNAME_KEY` environment variable.

Reply:
I think it would be great to drop use of these environment variables and require direct configuration, but preferable not as part of this PR.

Reply:
Yes, I agree! How about we open up an issue about dropping env vars support (if one isnt' open already) after merging this PR and we release them both together as a big breaking version?

Alternatively, I believe we could have the new `username_claim` track OAUTH2_USERNAME_KEY and default to that if not explicitly set as config option?

(Also, I believe there might be another similar situation with another env var)

Reply:
> I believe we could have the new username_claim track OAUTH2_USERNAME_KEY and default to that if not explicitly set as config option?

Let's go for this for now and either an inline FIXME note or dedicated issue on how this is something we should stop supporting this in the future.